scipy/sparse/linalg/_eigen/tests/test_svds.py

```diff
@@ -640,6 +647,24 @@ def test_svd_LM_zeros_matrix(self, shape, dtype):
         # Check that the singular values are zero.
         assert_array_equal(s, 0)
 
+    @pytest.mark.parametrize("shape", ((20, 20), (20, 21), (21, 20)))
+    @pytest.mark.parametrize("dtype", (float, complex, np.float32))
+    def test_small_sigma(self, shape, dtype):
+        if dtype == complex and self.solver == 'propack':
+            return
+        A = random_sample(shape).astype(dtype)
```

Comment:
We should have reproducible values in such tests.
```suggestion
        rng = np.random.default_rng(179847540)
        A = rng.random(shape, dtype= dtype)
```